src/main/java/com/inin/analytics/elasticsearch/ESEmbededContainer.java

```diff
@@ -127,65 +137,96 @@ public void deleteSnapshot(String snapshotName, String snapshotRepoName) {
 			.put("node.name", nodeName)
 			.put("path.data", workingDir)
-			.put("plugins." + PluginsService.LOAD_PLUGIN_FROM_CLASSPATH, true) // Allow plugins if they're bundled in with the uuberjar
-			.put("index.refresh_interval", -1) 
-			.put("index.translog.flush_threshold_size", "128mb") // Aggressive flushing helps keep the memory footprint below the yarn container max. TODO: Make configurable 
-			.put("bootstrap.mlockall", true)
-			.put("cluster.routing.allocation.disk.watermark.low", 99) // Nodes don't form a cluster, so routing allocations don't matter
-			.put("cluster.routing.allocation.disk.watermark.high", 99)
-			.put("index.load_fixed_bitset_filters_eagerly", false)
-			.put("indices.store.throttle.type", "none") // Allow indexing to max out disk IO
-			.put("indices.memory.index_buffer_size", "5%") // The default 10% is a bit large b/c it's calculated against JVM heap size & not Yarn container allocation. Choosing a good value here could be made smarter.
-			.put("index.merge.policy.max_merged_segment", MAX_MERGED_SEGMENT_SIZE_MB + "mb") // The default 5gb segment max size is too large for the typical hadoop node
-			//.put("index.merge.policy.max_merge_at_once", 10) 
-			.put("index.merge.policy.segments_per_tier", 4)
-			.put("index.merge.scheduler.max_thread_count", 1)
-			.put("path.repo", snapshotWorkingLocation)
-			.put("index.compound_format", false) // Explicitly disable compound files
-			//.put("index.codec", "best_compression") // Lucene 5/ES 2.0 feature to play with when that's out
-			.put("indices.fielddata.cache.size", "0%");
+            .put("path.repo", snapshotWorkingLocation)
+            .put("path.home", "/tmp")
+            .put("bootstrap.memory_lock", true)
+			.put("cluster.routing.allocation.disk.watermark.low", "99%") // Nodes don't form a cluster, so routing allocations don't matter
+			.put("cluster.routing.allocation.disk.watermark.high", "99%")
+            .put("node.data", true)       //data node
+            .put("transport.type", "local")
+            .put("indices.memory.index_buffer_size", "5%") // The default 10% is a bit large b/c it's calculated against JVM heap size & not Yarn container allocation. Choosing a good value here could be made smarter.
+            .put("indices.fielddata.cache.size", "0%")
+            .put("cluster.name", clusterName);
 			
-			if(memoryBackedIndex) {
-				builder.put("index.store.type", "memory");
-			}
 			Settings nodeSettings = builder.build();
 
-			// Create the node
-			container.setNode(nodeBuilder()
-					.client(false) // It's a client + data node
-					.local(true) // Tell ES cluster discovery to be inner-jvm only, disable HTTP based node discovery
-					.clusterName(clusterName)
-					.settings(nodeSettings)
-					.build());
+			try {
+                //read plugin list
+                ArrayList<Class<?>> pluginClasses = new ArrayList<Class<?>>();
```

Comment:
This method's getting a bit lengthy. Lets break gathering the plugin list into a separate method. Also, if there's a config file to specify plugins to be loaded you should document that optional bit in the readme file.